modules/compute/vm-instance/variables.tf

```
@@ -328,9 +328,20 @@ variable "placement_policy" {
 }
 
 variable "spot" {
-  description = "Provision VMs using discounted Spot pricing, allowing for preemption"
+  description = "DEPRECATED - Use `provisioning_model` instead."
   type        = bool
-  default     = false
+  default     = null
+
+  validation {
+    condition     = var.spot == null
+    error_message = "The 'spot' setting is deprecated. If you intend to use a Spot reservation please set 'var.provisioning_model' to 'SPOT' instead. Otherwise do not set the 'spot' setting at all"
```

Spot is not a reservation so just massaging the text a bit.

```suggestion
    error_message = "The 'spot' setting is deprecated. If you intend to use Spot pricing, please set 'var.provisioning_model' to \"SPOT\" and do not set 'var.spot' at all."
```